include/tvm/relay/op.h

```diff
@@ -348,23 +349,25 @@ inline OpRegistry& OpRegistry::add_argument(const std::string& name,
 
 inline OpRegistry& OpRegistry::add_type_rel(
     const std::string& rel_name,
-    std::function<Array<Type>(const Array<Type>&, int)> type_rel_func) {
+    runtime::TypedPackedFunc<bool(const Array<Type>&,
+                                  int,
+                                  const Attrs&,
+                                  const TypeReporter&)> type_rel_func) {
   auto func_name = std::string("tvm.relay.type_relation.") + rel_name;
-
-  TypedEnvFunc<Array<Type>(const Array<Type>&, int)> env_type_rel_func;
+  TypeRelationFn env_type_rel_func;
 
   if (runtime::Registry::Get(func_name)) {
     auto env_func = EnvFunc::Get(func_name);
     env_type_rel_func = env_func;
   } else {
     runtime::Registry::Register(func_name)
-        .set_body_typed<Array<Type>(const Array<Type>&, int)>(type_rel_func);
```

Comment:
Why not set the typed body here?

Reply:
There is no type checking here, so just to type less type signatures..